### PR TITLE
Make FrameGeometry::generateDecorations enforce hints.show_fames and …

### PR DESCRIPTION
**Pull request by aymanhab**
…update test case accordingly

### CODE REVIEW

**Comment by aymanhab** [comment]
This fixes the issue reported in PR #1403 and earlier where the flag to remove display of frames is ignored.
@tkuchida or @chrisdembia please take a look when you have a chance. Thanks. 🦃

**Comment by aymanhab** [comment]
Thanks for the feedback @chrisdembia the reason I did the change in FrameGeometry was for consistency (hints are used exclusively in generateDecorations rather than hidden a few levels deep) so it's easier to maintain, also hints will not be used for all other types of geometry so it will be confusing to pass them in and not use them except for FrameGeometry. In essence I wanted changes localized as much as possible to FrameGeometry that's why I implemented the method there.  What do you think?

**Comment by aymanhab** [comment]
Agreed that the test for connection/input in FrameGeometry::generateDecorations is redundant. Will remove.

File: OpenSim/Tools/Test/testVisualization.cpp [hunk 1]
```diff
@@ -429,35 +434,37 @@ void populate_contactModelPrimitives(SimTK::Array_<DecorativeGeometry>& stdPrimi
...
-        .setIndexOnBody(0).setScale(0.2).setOpacity(1)
-        .setRepresentation(SimTK::DecorativeGeometry::DrawSurface));
-    // Frames for the Joint
-    stdPrimitives.push_back(
-        DecorativeFrame(1.0).setBodyId(1).setColor(SimTK::White)
-        .setIndexOnBody(0).setScale(0.2).setOpacity(1)
-        .setRepresentation(SimTK::DecorativeGeometry::DrawSurface)
-        .setTransform(Vec3({-.01, -.03, .03})));
+    if (includeFrames) {
+        stdPrimitives.push_back(
```
Comment (tkuchida) [inline OpenSim/Tools/Test/testVisualization.cpp]: Perhaps cleaner to use a `for` loop here? …though I see that all you did was indent the existing code.

**Comment by aymanhab** [comment]
Thanks much @tkuchida 
Please let me know if I addressed your questions/suggestions @chrisdembia Thanks.

**Comment by chrisdembia** [comment]
> Please let me know if I addressed your questions/suggestions @chrisdembia Thanks.

Yes; thank you.

**Comment by aymanhab** [comment]
Thanks much @chrisdembia and @tkuchida Merging.

**Pull request merged**